ntoskrnl/ob/obname.c

```diff
@@ -713,7 +702,18 @@ ObpLookupObjectName(IN HANDLE RootHandle OPTIONAL,
                     (*(PULONGLONG)(ObjectName->Buffer) ==
                      ObpDosDevicesShortNamePrefix.Alignment.QuadPart))
                 {
-                    /* FIXME! */
+                    DeviceMap = ObpReferenceDeviceMap();
+                    /* We have a local mapping, drop the ?? prefix */
+                    if (DeviceMap != NULL && DeviceMap->DosDevicesDirectory != NULL)
+                    {
+                        LocalName = *ObjectName;
+
+                        LocalName.Length -= (4 * sizeof(WCHAR));
```

Comment:
What about decreasing the LocalName.MaximumLength too? (which represents the max byte size for the buffer pointed by LocalName.Buffer, which also appears to be changed just below)

Reply:
Agreed. Probably not technically necessary, but good practice.
Also, this should use `ObpDosDevicesShortName.Length`